zerver/lib/test_classes.py

```diff
@@ -113,6 +120,17 @@ def tearDown(self) -> None:
                 self.mock_ldap.reset()
             self.mock_initialize.stop()
 
+    def run(self, result: Optional[TestResult]=None) -> Optional[TestResult]:
+        if not settings.BAN_TEST_OUTPUT_EXTRA_SPAM:
+            return super(ZulipTestCase, self).run(result)
+        spam_finder = SpamFinder()
+        with TeeStderrAndFindSpam(spam_finder), TeeStdoutAndFindSpam(spam_finder):
+            test_result = super(ZulipTestCase, self).run(result)
+        if spam_finder.full_spam:
+            exception_message = f"\n\n--- SPAM DETECTED ---\n\nTest:\n{self.id()}\n\nSpam:\n{spam_finder.full_spam}\n\n ---------------"
```

Comment:
Let's format this using `"""` to be a more readable block.  And I'd replace "Spam"  with "Output" and "SPAM DETECTED" with "UNEXPECTED CONSOLE OUTPUT" and maybe some text about the goal of having `test-backend` output be clean.